pkg/compare/parsing.go

```diff
@@ -35,14 +35,28 @@ const (
 )
 
 type Component struct {
-	Name              string        `json:"name"`
-	Type              ComponentType `json:"type,omitempty"`
-	RequiredTemplates []string      `json:"requiredTemplates,omitempty"`
-	OptionalTemplates []string      `json:"optionalTemplates,omitempty"`
+	Name string        `json:"name"`
+	Type ComponentType `json:"type,omitempty"`
+	// Config 			  ReferenceTemplateConfig         `json:"config,omitempty"` // TODO figure out how to make this work
+	RequiredTemplates []*ReferenceTemplate `json:"requiredTemplates,omitempty"`
+	OptionalTemplates []*ReferenceTemplate `json:"optionalTemplates,omitempty"`
+}
+type ReferenceTemplateConfig struct {
+	AllowMerge bool `json:"allow-undefined-extras,omitempty"`
+}
+
+type ReferenceTemplate struct {
+	*template.Template
+	Path   string                  `json:"path"`
+	Config ReferenceTemplateConfig `json:"config,omitempty"`
```

Comment:
wondering why you want to leave the config section? i think it doesn't have some extra value and may be more annoying, like in Component the type is under component not under config    
anyway i don't have a string opinion so whatever you think is best

Reply:
I think its a good idea for config to have its own type, I think it will make things like allowing an ansible style hierarchical config easier. We can always remove it later if it turns out to be a pain.